packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,648 @@
+
+		// Draw the tickmarks as a patern over the gradient background
+		const tickmarksGradientdPattern = `0 center / calc((100% - ${tickmarkWidth}) / (${this._tickmarksAmount})) 100% repeat-x`;
+
+		// Combine to get the complete CSS background gradient property value
+		this._tickmarksBackground = `${tickmarksGradientBase + tickmarksGradientdPattern}`;
+
+		// If labelsInterval is specified draw labels for the necessary tickmarks
+		if (this.labelInterval > 0) {
+			this._drawDefaultLabels();
+		}
+	}
+
+	/**
+	 * Calculates the labels amout, width and text and creates them
+	 *
+	 * @private
+	 */
+	_drawDefaultLabels() {
+		const labelInterval = this.labelInterval;
+		const step = this.step;
+		const newNumberOfLabels = (this.max - this.min) / (step * labelInterval);
+
+		// If the required labels are already rendered
+		if (newNumberOfLabels === this._oldNumberOfLabels) {
+			return;
+		}
+
+		this._oldNumberOfLabels = newNumberOfLabels;
+		this._labelWidth = 100 / newNumberOfLabels;
+		this._labelItems = [];
+
+		// If the step value is not a round number get its precision
+		const stepPrecision = SliderBase._getDecimalPrecisionOfNumber(step);
+
+		// numberOfLabels below can be float so that the "distance betweenlabels labels"
+		// calculation to be precize (exactly the same as the distance between the tickmarks).
+		// That's ok as the loop stop condition is set to an integer, so it will practically
+		// "floor" the number of labels anyway.
+		for (let i = 0; i <= newNumberOfLabels; i++) {
+			// Format the label numbers with the same decimal precision as the value of the step property
+			const labelItemNumber = ((i * step * labelInterval) + this.min).toFixed(stepPrecision);
+			this._labelItems.push(labelItemNumber);
+		}
+	}
+
+	/**
+	 * Calculates space between tickmarks
+	 *
+	 * @private
+	 */
+	_spaceBetweenTickmarks() {
+		const tickmarksAmountStrCalc = this._tickmarksAmount.split("/");
+		const tickmarksAmount = tickmarksAmountStrCalc[0] / tickmarksAmountStrCalc[1];
+
+		return this.getBoundingClientRect().width / tickmarksAmount;
+	}
+
+	/**
+	 * Normalizes a new <code>step</code> property value.
+	 * If tickmarks are enabled recreates them according to it.
+	 *
+	 * @private
+	 */
+	_setStep(step) {
+		if (step === 0) {
+			console.warn("The 'step' property must be a positive float number"); // eslint-disable-line
+			return;
+		}
+
+		if (step < 0) {
+			console.warn("The 'step' property must be a positive float number. The provided negative number has been converted to its positve equivalent"); // eslint-disable-line
+			step = Math.abs(step);
+		}
+
+		if (typeof step !== "number" || Number.isNaN(step)) {
+			console.warn("The 'step' property must be a positive float number. It has been set to its default value of 1"); // eslint-disable-line
+			step = 1;
+		}
+
+		if (!this._initialRendering) {
+			this._drawDefaultTickmarks(step, this.max, this.min);
+		}
+
+		this.step = step;
```

Comment:
Same thing as min and max. We cannot change what the app provided. Please use `get _effectiveStep()` everywhere in the code instead of `this.step`